var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -50,13 +50,22 @@ class Legion(CMakePackage):
     # TODO: we could use a map here to clean up and use naming vs. numbers.
     cuda_arch_list = ('60', '70', '75', '80')
     for nvarch in cuda_arch_list:
-        depends_on('kokkos@3.3.01+cuda+cuda_lambda+wrapper cuda_arch={0}'.format(nvarch),
+        depends_on('kokkos@3.3.01:+cuda+cuda_lambda+wrapper cuda_arch={0}'.format(nvarch),
                    when='%gcc+kokkos+cuda cuda_arch={0}'.format(nvarch))
-        depends_on("kokkos@3.3.01+cuda+cuda_lambda~wrapper cuda_arch={0}".format(nvarch),
+        depends_on("kokkos@3.3.01:+cuda+cuda_lambda~wrapper cuda_arch={0}".format(nvarch),
                    when="%clang+kokkos+cuda cuda_arch={0}".format(nvarch))
 
-    depends_on('kokkos@3.3.01~cuda', when='+kokkos~cuda')
-    depends_on("kokkos@3.3.01~cuda+openmp", when='+kokkos+openmp')
+    depends_on('kokkos@3.3.01:~cuda', when='+kokkos~cuda')
+    depends_on("kokkos@3.3.01:~cuda+openmp", when='+kokkos+openmp')
+
+    # HIP specific
+    variant('hip_hijack', default=False,
+            description="Hijack application calls into the HIP runtime (+rocm).")
+    variant('hip_target', default='ROCM',
+            values=('ROCM', 'CUDA'),
+            description="API used by HIP",
+            multi=False)
+    depends_on('kokkos@3.3.01:+rocm', when='+kokkos+rocm')
```

Comment:
Does `legion +rocm amdgpu_target=X` require the same amdgpu_target from `kokkos +rocm` ? If so, that could be expressed the same way `cuda_arch` is passed to the kokkos dependency a few lines above.

Reply:
@eugeneswalker pushed a change that propagates the `amdgpu_target`